remote-api/src/main/java/org/alfresco/rest/api/impl/SizeDetailsImpl.java

```diff
@@ -0,0 +1,132 @@
+import org.alfresco.util.GUID;
+import org.alfresco.util.ParameterCheck;
+import org.springframework.beans.factory.InitializingBean;
+
+public class SizeDetailsImpl implements SizeDetails, InitializingBean
+{
+    private Nodes nodes;
+    private NodeRef nodeRef;
+    private NodeSizeDetailsService nodeSizeDetailsService;
+
+    public void setNodes(Nodes nodes)
+    {
+        this.nodes = nodes;
+    }
+
+    public void setNodeSizeDetailsService(NodeSizeDetailsService nodeSizeDetailsService)
+    {
+        this.nodeSizeDetailsService = nodeSizeDetailsService;
+    }
+
+    /**
+     * generateNodeSizeDetailsRequest : providing HTTP STATUS 202 with jobId.
+     */
+    @Override
+    public NodeSizeDetails generateNodeSizeDetailsRequest(String nodeId)
+    {
+        nodeRef = nodes.validateOrLookupNode(nodeId);
+        validateType(nodeRef);
+        String actionId;
+        if (!nodeSizeDetailsService.checkSizeDetailsExist(nodeId))
+        {
+            actionId = executeSizeDetails();
+        }
+        else
+        {
+            NodeSizeDetails nodeSizeDetails = nodeSizeDetailsService.getSizeDetails(nodeId);
+            actionId = nodeSizeDetails.getJobId();
+        }
+        return new NodeSizeDetails(actionId);
+    }
+
+    /**
+     * getNodeSizeDetails : providing HTTP STATUS 200 with NodeSizeDetails data from cache.
+     */
+    @Override
+    public NodeSizeDetails getNodeSizeDetails(final String nodeId, final String jobId)
+    {
+        NodeRef nodeRef = nodes.validateOrLookupNode(nodeId);
+        validateType(nodeRef);
+
+        if (!nodeSizeDetailsService.checkSizeDetailsExist(nodeId))
+        {
+            NodeSizeDetails nodeSizeDetails = new NodeSizeDetails(nodeId,STATUS.NOT_INITIATED);
+            return nodeSizeDetails;
+        }
+        else
+        {
+            NodeSizeDetails nodeSizeDetails = nodeSizeDetailsService.getSizeDetails(nodeId);
+            String cachedJobId = nodeSizeDetails.getJobId();
+            if (cachedJobId != null && !jobId.equalsIgnoreCase(cachedJobId))
+            {
+                throw new NotFoundException("jobId does not exist");
+            }
+        }
+
+        return nodeSizeDetailsService.getSizeDetails(nodeId);
+    }
+
+    /**
+     * Executing Asynchronously.
+     */
+    private String executeSizeDetails()
+    {
+        String jobId = GUID.generate();
+        NodeSizeDetails nodeSizeDetails = new NodeSizeDetails(nodeRef.getId(), jobId, STATUS.PENDING);
+        nodeSizeDetailsService.putSizeDetails(nodeRef.getId(), nodeSizeDetails);
+        nodeSizeDetailsService.invokeSizeDetailsExecutor(nodeRef, jobId);
+        return jobId;
+    }
+
+    private void validateType(NodeRef nodeRef) throws InvalidNodeTypeException
+    {
+        if (!nodes.isSubClass(nodeRef, ContentModel.TYPE_FOLDER, false))
+        {
+            throw new InvalidNodeTypeException("Invalid parameter: value of nodeId is invalid");
```

Comment:
The exception message could be more informative. Consider changing it to `Node id " + nodeId + " does not represent a folder.`

Reply:
As suggested,
Updated.